### DISCUSSION TITLE
本科的校外毕设的信息录入指令报错

### DISCUSSION THREAD

**Discussion by MrZLeo** [note286/xduts#124] [Q&A] [answered]
xduts的手册有说校外和校内导师两个类型，填入`supv-ent`和`supv-school`都会编译报错：
```latex
\xdusetup{
	info = {
		% supv-ent = {xxx},
		% supv-school = {aaa}
		supervisor = {bbb}, % 这行不会报错
	}
}
```

**Comment by note286** [comment]
请使用最新版的`.cls`。

**Comment by Kiki2049** [comment]
我刚刚更新版本为v6.1.4.2（2023/03/23)，仅使用`supv-ent`和`supv-school`:

```
\xdusetup{
	info = {
		 supv-ent = {xxx},
		 supv-school = {xx}
	}
}
```
但生成的页面还是显示了`导师姓名`：
![image](https://user-images.githubusercontent.com/55939997/236137857-a7316607-4f15-4df3-bdbd-072b4e16cc8d.png)

**Reply by note286** [reply, marked as answer]
由于一些原因，暂时不方便提交新版本，你手动修改一下吧，是我在2022-05-21的时候不小心引入的bug。

你将`xduugthesis.cls`中第`1666`行的`\group_begin:`改为`{`，将`1676`行的`\group_end:`改为`}`。

**Reply by Kiki2049** [reply]
感谢！

**Reply by note286** [reply]
顺便问问，有校外导师的又用xduts的人应该没几个吧，暂时不方便提交新版本，如果有人有问题就先手动修改下吧。

**Reply by Kiki2049** [reply]
这个不太清楚，如果有人有这个问题我会告诉他们看这个discussion的

**Reply by note286** [reply]
我已经置顶了。

**Comment by note286** [comment]
@Dttt-chowhound 仅作如下两处修改即可，其他不要动，将修改后的`xduugthesis.cls`放置于与`.tex`文件同一目录中。
![image](https://github.com/user-attachments/assets/408e95ff-f475-48b3-9dc6-b53142333776)

```latex
\documentclass{xduugthesis}
\xdusetup{
info = {
supv-ent = {校外},
supv-school = {校内}
}}
\begin{document}
\end{document}
```

**Reply by ghost** [reply]
好的，可能是因为我使用的是texlive2024，它使用的是内置的xduts，我再试试，感谢

**Reply by note286** [reply]
你是没法直接修改TeX Live目录下的文件的，是只读的，你拷贝出来再修改并放置于与`.tex`文件同一目录中。

```
C:\texlive\2024\texmf-dist\tex\xelatex\xduts\xduugthesis.cls
```

**Reply by ghost** [reply]
感谢，该问题已解决！

**Comment by note286** [comment]
https://github.com/note286/xduts/commit/938acd70c2cc98c658ffd0523288bbd5bd8d19a2 已修复。
